Rationale.md

```diff
@@ -162,12 +173,13 @@ variables by creating a separate stack data structure within linear memory. This
 stack is sometimes called the "aliased" stack, since it is used for variables
 which may be pointed to by pointers.
 
-This prevents WebAssembly from performing clever optimizations on the stack and
-liveness of such variables, but this loss isn't expected to be
-consequential. Common C compiler optimizations such as LLVM's global value
-numbering effectively split address-taken variables into parts, shrinking the
-range where they actually need to have their address taken, and creating new
-ranges where they can be allocated as local variables.
+Since the aliased stack appears to the WebAssembly engine as normal memory,
+WebAssembly optimizations that would target the aliased stack need to be more
+general, and thus more complicated. We observe that common compiler
+optimizations on the producer side, such as LLVM's global value numbering,
```

Comment:
Do we ever define "producer"? It is also mentioned in AstSemantics.md but I don't think it is ever defined. It is pedantic but these kind of docs usually are. Perhaps it should be defined in README.md or a separate Terminology doc.